src/main/scala/is/hail/HailContext.scala

```diff
@@ -523,7 +523,7 @@ class HailContext private(val sc: SparkContext,
     maybeGZipAsBGZip(forceBGZ) {
       TextTableReader.read(this)(files, types, comment, separator, missing,
         noHeader, impute, nPartitions.getOrElse(sc.defaultMinPartitions), quote,
-        skipBlankLines).keyBy(keyNames.map(_.toArray), sort = false)
+        skipBlankLines).keyBy(keyNames.map(_.toArray))
```

Comment:
Yeah, this is confusing. The semantics of the `sort` parameter changed completely. It used to mean "if `sort=false` then produce something which isn't sorted". Now tables are always sorted by key. Now the semantics is "if `sort=false`, then the builder of the `TableKeyBy` node promises that the table will still be in key order after the map".

So in this case, we used to let this produce an unsorted table. Now we can't do that, so we have to allow the `TableKeyBy` to do a scan, and if necessary a shuffle, by setting `sort=true`.

I don't love the name `sort` for this parameter, but never thought of anything better.

Reply:
How about `isSorted`?

Reply:
Yeah, that's better. I made the change on `TableKeyBy`, and also on `Table.keyBy` and `Table.apply` for consistency.

Reply:
I'm a little worried that there could be instances of `sort = false` missed that need to be flipped to true. Did you do a search for `sort =`?

Reply:
That wouldn't even catch everything, because it's not always called with a keyword argument. I was careful to look at every call site (there aren't that many, and most of them use the default value).